### ISSUE TITLE
Wyze-bridge stops retrying to connect to cameras after they've been off for a while

### ISSUE THREAD

**Issue by jappish84** [mrlt8/docker-wyze-bridge#306]
I recently updated the wyze-bridge docker container and am now running version 1.27.4. 

I noticed that wyze-bridge stops retrying to reconnect to offline cameras and only chooses a single camera to retry reconnecting to, all other cameras seem to get dropped. This was not the case with previous versions where camera streems would become available shortly after the cameras were turned on after being off. Simly restarting the container temporarly solves the issues and all cameras come online again.

I have a total of 6 cameras, here is what the logs look like, notice "uppfarten" camera is the only camera beeing retried:

```
2022/02/26 08:53:33 [WyzeBridge] 👻 Uppfarten offline. WILL retry in 10s.
2022/02/26 08:53:43 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Uppfarten on 192.168.1.190 (1/3)
2022/02/26 08:53:44 [Uppfarten] IOTC_ER_DEVICE_OFFLINE
2022/02/26 08:53:45 [WyzeBridge] 👻 Uppfarten offline. WILL retry in 10s.
2022/02/26 08:53:55 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Uppfarten on 192.168.1.190 (1/3)
2022/02/26 08:53:56 [Uppfarten] IOTC_ER_DEVICE_OFFLINE
2022/02/26 08:53:57 [WyzeBridge] 👻 Uppfarten offline. WILL retry in 10s.
2022/02/26 08:54:07 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Uppfarten on 192.168.1.190 (1/3)
2022/02/26 08:54:08 [Uppfarten] IOTC_ER_DEVICE_OFFLINE
2022/02/26 08:54:09 [WyzeBridge] 👻 Uppfarten offline. WILL retry in 10s.
2022/02/26 08:54:19 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Uppfarten on 192.168.1.190 (1/3)
2022/02/26 08:54:22 [Uppfarten] 📡 Getting 120kb/s HD stream via LAN mode (WiFi: 84%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/02/26 08:54:26 [RTSP][UPPFARTEN] ✅ '/uppfarten' stream is UP! (3/3)
2022/02/26 08:54:26 [RTSP][UPPFARTEN] 📖 New client reading 
```

Edit: note, I have the `IGNORE_OFFLINE` env var set to `False`

**Comment by mrlt8** [comment]
I've noticed something similar. It seems like the process is not exiting correctly, so it doesn't gets restarted.

**Comment by jappish84** [comment]
That was my initial though aswell, so I updated most cams and later saw that the one cam that did get updated was a cam on old fw, the next day it would be the new fw, so seems random to me.

The only pattern I can see is that the program only ever tries a single camera, I.e the logs look like the one in the original post, only it would be a different camera each time.


Not sure if this maybe helps, but I have an automation that calls the turn off command for the cameras when we are at home. The automation turns on outside cams in the evening and turns all cams on if we are away so all the cams will be toggling on/off a few times a day

**Comment by jappish84** [comment]
Thanks @mrlt8 I just updated, will test and report back 👍

**Comment by jappish84** [comment]
I'd say you did a great job! Seems like the issue is done 🎉🎉

Thank you 🙏

**Comment by mrlt8** [comment]
Awesome! I'll mark this issue as resolved for now, but keep an eye out for a related issue #319 where the bridge could go into a loop if the connection times out.

**Comment by jappish84** [comment]
Hmm, seems there's still some issue, just noticed that one of my cams has dropped out of the connection retry loop, cam has been on for a few hours, but not getting picked up by wyze-bridge. Other cameras that are currently offline are periodically being checked

**Comment by mrlt8** [comment]
Anything unusual in the logs? Did that cam stop trying after step 1/3?

**Comment by jappish84** [comment]
Nothing else unusual, but that specific cam is totally gone from the logs, no attempt at reconnection, as if it's gone from the account, a restart of the container solves it temporarily

**Comment by mrlt8** [comment]
Does it keep happening with that one camera?

**Comment by jappish84** [comment]
Not sure yet, let me test it for a bit longer and I'll report back

**Comment by jappish84** [comment]
Ok, so today I discovered that any form of reconnection attempt seemed to have stopped. All cams were started at 1630 o'clock, but the logs are frozen at 10.01 o'clock. Checking date in the console reports the correct date.

Restarting the container once again solves the issue temporarily

**Comment by mrlt8** [comment]
So is the bridge getting stuck while the cameras are online or while they're still offline?

**Comment by jappish84** [comment]
They got stuck while all cams were offline

**Comment by mrlt8** [comment]
Thanks, I'll try to replicate the issue!

**Comment by jappish84** [comment]
Hmm... I haven't done anything different, but I see this behavior now where wyze-bridge tries to connect to just two of the cameras. All cams are on, but connection through wyze-bridge fails to all cams.

```
2022/03/17 07:47:07 [Vardagsrum] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 84%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/03/17 07:47:08 [ovanvaning] [Errno 32] Broken pipe
2022/03/17 07:47:09 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - ovanvaning on 192.168.1.204 (1/3)
2022/03/17 07:47:11 [Vardagsrum] [Errno 32] Broken pipe
2022/03/17 07:47:12 [ovanvaning] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 91%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/03/17 07:47:12 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Vardagsrum on 192.168.1.85 (1/3)
2022/03/17 07:47:14 [Vardagsrum] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 85%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/03/17 07:47:18 [ovanvaning] [Errno 32] Broken pipe
2022/03/17 07:47:19 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - ovanvaning on 192.168.1.204 (1/3)
2022/03/17 07:47:19 [Vardagsrum] [Errno 32] Broken pipe
2022/03/17 07:47:20 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Vardagsrum on 192.168.1.85 (1/3)
2022/03/17 07:47:21 [ovanvaning] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 87%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/03/17 07:47:22 [Vardagsrum] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 84%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/03/17 07:47:26 [ovanvaning] [Errno 32] Broken pipe
2022/03/17 07:47:26 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - ovanvaning on 192.168.1.204 (1/3)
2022/03/17 07:47:27 [Vardagsrum] [Errno 32] Broken pipe
2022/03/17 07:47:28 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Vardagsrum on 192.168.1.85 (1/3)
2022/03/17 07:47:28 [ovanvaning] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 89%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/03/17 07:47:30 [Vardagsrum] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 83%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/03/17 07:47:34 [ovanvaning] [Errno 32] Broken pipe
2022/03/17 07:47:34 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - ovanvaning on 192.168.1.204 (1/3)
2022/03/17 07:47:35 [Vardagsrum] [Errno 32] Broken pipe
2022/03/17 07:47:36 [WyzeBridge] 🎉 Connecting to WyzeCam V2 - Vardagsrum on 192.168.1.85 (1/3)
2022/03/17 07:47:36 [ovanvaning] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 88%) FW: 4.9.8.501 🔒 (DTLS) (2/3)
2022/03/17 07:47:39 [Vardagsrum] 📡 Getting 120kb/s HD stream @15fps via LAN mode (WiFi: 83%) FW: 4.9.8.501 🔒 (DTLS) (2/3)


```

Again, restarting the container resolves all issues temporarily

**Comment by mrlt8** [comment]
Changes have been merged into v1.3.0.

Please let me know if this helps.

**Comment by jappish84** [comment]
Thank you! I have updated the container, will test and report back
